nilearn/experimental/surface/maskers.py

```diff
@@ -608,9 +609,7 @@ def inverse_transform(self, masked_img: np.ndarray) -> SurfaceImage:
 
     def generate_report(self):
         """Generate a report."""
-        try:
-            from nilearn.reporting.html_report import generate_report
-        except ImportError:
+        if not is_matplotlib_installed():
             with warnings.catch_warnings():
                 mpl_unavail_msg = (
                     "Matplotlib is not imported! "
```

Comment:
Could this warning be part of `is_matplotlib_installed`?

Reply:
actually I reverted this change because it would not raise the proper error when matploblib was not installed